src/Lecture1.hs

```diff
@@ -104,7 +126,11 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: [Char] -> Int
+strSum str = sum (map stringToInt (words str))
+  where
+    stringToInt :: [Char] -> Int
+    stringToInt str = read str :: Int
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
    stringToInt str = read str
```

You can even write `strSum` as a one-liner 🙂